config/config.go

```diff
@@ -44,12 +46,37 @@ type GinkgoConfigType struct {
 var GinkgoConfig = GinkgoConfigType{}
 
 type DefaultReporterConfigType struct {
-	NoColor           bool
 	SlowSpecThreshold float64
 	NoisyPendings     bool
 	Succinct          bool
 	Verbose           bool
 	FullTrace         bool
+
+	color   bool
+	noColor bool
+
+	// Colorize caches whether or not we want to colorize output
+	Colorize *bool
+}
+
+// ColorizeOutput determines if output should be colorized. The user
+// can specify they want color (--color) or not (--noColor). Further-
+// more, we detect if the output to which we are printing is a TTY.
+// If the output is a TTY, the default is to use color but a user can
+// still request no color; if the output is not a TTY, the default is
+// to not use color but the user can still request it.
+func (c *DefaultReporterConfigType) ColorizeOutput() bool {
+	if c.Colorize == nil {
+		var colorize bool
+		if terminal.IsTerminal(int(os.Stdout.Fd())) {
```

Comment:
There's nothing that necessarily ties the default reporter to `stdout`, so technically this may be incorrect. I can refactor to accept to take the output stream as a parameter if that is more appropriate.